### PR TITLE
Retina api 109

### PR DESCRIPTION
**Pull request by ekraihan**
Ok boys, your one-stop-shop for tool snapshot searching

### CODE REVIEW

File: tests/test-db.js [hunk 1]
```diff
@@ -1029,6 +1077,6 @@ describe('Database creation and usage', async () => {
 
   after(async () => {
     await dbClient.disconnect();
-    await dropDbIfExists(appConfig['db.database']);
+    // await dropDbIfExists(appConfig['db.database']);
```
Comment (Jbcampbe) [inline tests/test-db.js]: ☝️

File: graphql/schema/tool-history.graphql [hunk 2]
```diff
@@ -1,41 +1,31 @@
...
-  purchased_from: ConfigurableItem
-  date_purchased: Date
-  photo:          String
-  price:          Int
-  year:           Int
+type ToolSnapshot {
+  id:                     ID!
+  tool:                   Tool!
+  metadata:               ToolSnapshotMetadata!
+  previous_tool_snapshot: ToolSnapshot
```
Comment (ekraihan) [inline graphql/schema/tool-history.graphql]: A tool snapshot consists of a tool (just a normal tool object), metadata (defined below), and the tool snapshot that came previous in time to this tool snapshot. This value can be null when there was not previous tool snapshot (like on a CREATE)

File: graphql/schema/tool-history.graphql [hunk 3]
```diff
@@ -1,41 +1,31 @@
...
-  metadata: ToolHistoryEntryMetadata!
-}
-
-input ToolHistoryFilter {
-  tool_ids:     [ID!]
-  owner_ids:    [ID!]
-  tool_actions: [ToolAction!]
-  time_span:    TimeSpan
+input ToolSnapshotFilter {
+  only_latest_snapshot: Boolean
```
Comment (ekraihan) [inline graphql/schema/tool-history.graphql]: @amenders this is the magic parameter that will only search over the latest tool snapshot for each tool

File: graphql/schema/tool-history.graphql [hunk 4]
```diff
@@ -1,41 +1,31 @@
...
+  brand_ids:            [ID!]
+  type_ids:             [ID!]
+  owner_ids:            [ID!]
+  tool_statuses:        [ToolStatus!]
 }
 
 type Query {
-  searchToolHistoryEntry(toolHistoryFilter: ToolHistoryFilter, pagingParameters: PagingParameters): [ToolHistoryEntry!]
+  getToolSnapshot(tool_snapshot_id: ID!): ToolSnapshot
+  searchToolSnapshot(query: String, toolSnapshotFilter: ToolSnapshotFilter, pagingParameters: PagingParameters): [ToolSnapshot!]
```
[the remark this answers is not in the document]
Comment (Jbcampbe) [reply graphql/schema/tool-history.graphql]: 👍

File: sql/triggers/tool-snapshot-created.sql [hunk 5]
```diff
@@ -0,0 +1,33 @@
+
+
+CREATE FUNCTION upsert_latest_tool_snapshot () RETURNS TRIGGER
+AS $$
+  BEGIN
+
+    UPDATE tool_snapshot
```
Comment (ekraihan) [inline sql/triggers/tool-snapshot-created.sql]: Update the just-inserted tool snapshot with the tool_snapshot_id that is currently in latest_tool_snapshot

**Pull request merged**